roxctl/deployment/check/check.go

```diff
@@ -123,12 +127,18 @@ func Command(cliEnvironment environment.Environment) *cobra.Command {
 	utils.Must(c.Flags().MarkDeprecated("print-all-violations", "use the new output format where all "+
 		"violations are printed by default. This flag will only be relevant in combination with the --json flag"))
 
+	// We need a string parameter with the filenames for the sarif printer factory
+	// If a report is based on a single yaml file it will work as previously, if multiple
+	// they will be concatenated as csv
+	deploymentCheckCmd.joinedFiles = strings.Join(deploymentCheckCmd.files, ",")
+
 	return c
 }
 
 type deploymentCheckCommand struct {
 	// properties bound to cobra flags
-	file               string
+	joinedFiles        string
```

Comment:
Fair points, and I'm a bit out of my depth when it comes to our Sarif printers, so great if Merlin can handle that, however.
> use the first image in the list of files instead. It'd potentially be a "wrong" reference, but at the very least not a invalid reference.
This is actually the first workaround I tried but it does not work since `deploymentCheckCmd.files` is empty at the time it is passed to `NewSarifPrinterFactory`, so pointing at the first element of the slice is out of range. I could create a `deploymentCheckCmd.firstFile` of type string and point at that, then populate it after flags are set as a workaround, what do you think?

Reply:
I've gone ahead and created the issue for SARIF printers https://issues.redhat.com/browse/ROX-21443
TODO @stackrox/merlin